src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -0,0 +1,511 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Collections;
+using System.Diagnostics;
+using System.Globalization;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using System.IO.Ports;
+using System.Text;
+
+namespace System.IO.Ports
+{
+    internal sealed partial class SerialStream : Stream
+    {
+        private const int infiniteTimeoutConst = -2;
+        private const int MaxDataBits = 8;
+        private const int MinDataBits = 5;
+
+        private string _portName;
+        private int _baudRate;
+        private StopBits _stopBits;
+        private Handshake _handshake;
+        private Parity _parity;
+        private int _dataBits = 8;
+        private bool _inBreak = false;
+        private SafeFileHandle _handle = null;
+        private bool _rtsEnable = false;
+        ArrayList selectList;
```

Comment:
naming, missing underscore, and why use ArrayList? Is there some usage that requires it to be an ArrayList?